Browser/browser.py

```diff
@@ -957,14 +965,58 @@ def _start_suite(self, name, attrs):
             except ConnectionError as e:
                 logger.debug(f"Browser._start_suite connection problem: {e}")
 
-    def _start_test(self, name, attr):
+    def _start_test(self, name, attrs):
+        self.timeout_stack.start(attrs["id"], Scope.Test)
         if self._auto_closing_level == AutoClosingLevel.TEST:
             try:
                 self._execution_stack.append(self.get_browser_catalog())
             except ConnectionError as e:
                 logger.debug(f"Browser._start_test connection problem: {e}")
 
+    def _start_keyword(self, name, attrs):
```

Comment:
This code is just moved and unaltered from the lower part of that file, so that it is in a better place between start  and end test